apps/coordinator/ts/deployer/deployer.service.ts

```diff
@@ -767,6 +892,64 @@ export class DeployerService {
     return { address: maciAddress };
   }
 
+  async deployMaci({ approval, sessionKeyAddress, chain, config }: IDeployMaciArgs): Promise<{ address: string }> {
+    const kernelClient = await this.sessionKeysService.generateClientFromSessionKey(sessionKeyAddress, approval, chain);
+    const signer = await this.sessionKeysService.getKernelClientSigner(kernelClient);
+
+    const policyContract = await this.getOrDeployPolicy(signer, config.policy.type, chain, config.policy.args);
+    const policyAddress = await policyContract.getAddress();
+
+    const verifierFactory = new VerifierFactory(signer);
+    const verifierContract = await verifierFactory.deploy();
+
+    const vkRegistryAddress = await deployVkRegistryContract({ signer });
+
+    const [verifyingKeysArgsQV, verifyingKeysArgsNOQV] = await Promise.all([
+      this.getVerifyingKeysArgs(signer, vkRegistryAddress as Hex, config.VkRegistry.args, EMode.QV),
+      this.getVerifyingKeysArgs(signer, vkRegistryAddress as Hex, config.VkRegistry.args, EMode.NON_QV),
+    ]);
+    await setVerifyingKeys(verifyingKeysArgsQV);
+    await setVerifyingKeys(verifyingKeysArgsNOQV);
```

Comment:
I am having a problem here: we are supposed to deployMaci with the QV and NO_QV right? I tried to run `setVerifyingKeys` twice but I am getting the error:

```
"This poll verifying key is already set in the contract"
```


I noticed that in the tests https://github.com/privacy-scaling-explorations/maci/blob/b4e6be4db2d884263e31233eabdd211ffb2b5342/packages/contracts/tests/VkRegistry.test.ts#L109 there are 4 scenarios:

1. should set the process, tally vks
2. should throw when trying to set another vk for the same params
3. should allow to set vks for different params
4.  should allow to set vks for different modes


Should scenario 4 works also in the coordinator? I am not sure why is it failing

Reply:
Should a MACI contract work with QV and NO_QV or only with one option? 🤔

Reply:
it should work with both, maybe something wrong in the sdk will check